src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -55,27 +55,62 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 	}
 
 	/**
-	 * Checks that a PID string is either null, or has at least one non-whitespace
-	 * character and conforms to the specified regular expression.
+	 * Trims leading and trailing whitespace and then checks that a string is either null,
+	 * or has at least one non-whitespace character and conforms to the specified regular
+	 * expression. If replace is not null, it is used for a replaceAll operation, and the
+	 * resulting string returned. Otherwise, the trimmed string is returned.
 	 *
-	 * @param putativePid the string to check.
-	 * @param name        the name of the string to use in any error messages.
-	 * @param optional    whether or not the field is optional. If true, null is a valid value for the PID.
-	 * @return the trimmed PID.
+	 * @param stringToCheck the string to check.
+	 * @param pattern       the pattern to validate against.
+	 * @param replace       if non-null, the pattern to use for the replaceAll operation.
+	 * @param name          the name of the string to use in any error messages.
+	 * @param optional      whether or not the field is optional. If true, null is a valid value for the field.
```

Comment:
Nit: null or whitespace

Reply:
> Trims leading and trailing whitespace and then checks that a string is either null, or has at least ...

Whitespace isn't a valid value; null is.

Reply:
Unless I'm completely misreading something here, if optional is true both `null` and `\t   \n` will result in this function returning `null`... right?

Reply:
Maybe the confusion is over value vs. input, I was thinking input

Reply:
Yup, that's it. How about this wording for the param:

```
	 * @param optional      whether or not the field is optional. If false, null,
         *                      empty, or whitespace-only input strings will throw an error.
```

Reply:
:+1: